### PR TITLE
Support branches for git dependencies

### PR DESCRIPTION
**Pull request by steveeJ**

### CODE REVIEW

File: src/template.rs [hunk 1]
```diff
@@ -134,13 +145,21 @@ fn to_source(pkg: &ResolvedPackage<'_>, cwd: &Path) -> Result<Source> {
                 .ok_or(anyhow!("checksum is required for crates.io package {}", id))?,
         }
     } else if id.source_id().is_git() {
+        let branch = if let Some(cargo::core::source::GitReference::Branch(branch)) =
```
Comment (edude03) [inline src/template.rs:148]: Minor improvement idea,

why not add `source::GitReference` to `use cargo::core::{...}`

then change this to GitReference::Branch?

File: src/template.rs [hunk 2]
```diff
@@ -90,10 +90,21 @@ pub struct Crate {
 
 #[derive(Debug, Serialize)]
 pub enum Source {
-    CratesIo { sha256: String },
```
Comment (psionic-k) [inline src/template.rs:93]: Prefer not to bleed whitespace changes in. Probably disable your auto format. I'll add a rustfmt configuration if we don't have one

:shipit:  Let's merge this, with or without whitespace. Ping me for re-approval if necessary

**Review by psionic-k — APPROVED** [review]

File: src/template.rs (hunk 2 above)
Comment (steveeJ) [reply src/template.rs:93]: @psionic-k :ping_pong:

**Review by psionic-k — APPROVED** [review]
:1st_place_medal:

**Pull request merged**